Issues with copyright and shebang validation when empty spaces after shebang.

# Summary
When we have a file with shebang, _header_ rule only accepts a copyright notice as valid only when it starts immediately on the line after the shebang - no empty lines are tolerated.

What is making this worse is that the fixer for missing header inserts the header just before the first line of meaningful code (with a number empty lines as defined in the options). This means that a file with at least one empty line between the shebang and the first line of code or comment will:
- never pass lint.
- when the issue gets fixed, the code will still not pass lint.

# Steps to Reproduce

Let's have the following ESLint config:

_eslint.config.mjs_ (snippet):
```js
        rules: {
            'header/header': [
                'error',
                'line',
                [
                    {
                        pattern: 'Copyright \\(c\\) \\d{4}-present John Doe',
                        template: 'Copyright (c) 2025-present John Doe',
                    },
                ],
            ]
        }
```

# Working Scenario

1. I lint the code to see a missing header:
```
# npx eslint .

.../index.js
  2:1  error  missing header  header/header

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```
2. I fix the header:
```
# npx eslint . --fix
```
_index.js_ (modified):
```js
#!/usr/bin/env node
//Copyright (c) 2025-present John Doe
console.log('test');
```
3. Just to be sure I run lint again with no errors.
```
# npx eslint . 
```
4. I can repeat steps 2 and 3 multiple times without any changes going into _index.js_.

# Failing Scenario

If we have an empty line after the shebang, lint fails but the `--fix` adds the copyright just in front of the code, and not after the shebang:

_index.js_ (initial):
```js
#!/usr/bin/env node

console.log('test');
```
1. Confirm there is a lint error:
```
# npx eslint .

.../index.js
  3:1  error  missing header  header/header

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```
2. Fixing the issue:
```
# npx eslint . --fix
```
_index.js_ (modified):
```js
#!/usr/bin/env node

//Copyright (c) 2025-present John Doe
//Copyright (c) 2025-present John Doe
//Copyright (c) 2025-present John Doe
//Copyright (c) 2025-present John Doe
//Copyright (c) 2025-present John Doe
//Copyright (c) 2025-present John Doe
//Copyright (c) 2025-present John Doe
//Copyright (c) 2025-present John Doe
//Copyright (c) 2025-present John Doe
//Copyright (c) 2025-present John Doe
console.log('test');
```

The console version of ESLint bails out after ten attempts to fix the issue. VS Code version bails out immediately after attempting the fix. In any case we come to a file that is still failing lint:

```
# npx eslint .   

.../index.js
  13:1  error  missing header  header/header

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```
As we attempt to fix the issue multiple times, the number of "invalid" copyright comments multiplies, but the lint keeps failing.

# Short-term Fix

For the next minor version, the proposed solution is to have the fix always match what the rule accepts as valid, that is, a copyright header straight after the shebang with no empty line in between.

# Long-term Fix

It is reasonable to support one or more mandatory or permissible empty lines between the shebang and the header. This needs to be configurable. Alas, the current design of the options is reaching its maximum in terms of both understandability and ease of parsing / maintenance (there was already a bug with line numbers and line ending options clashing - see #2). That's why for v4 I want to overhaul the options and have a more extensible and easier to maintain object-based schema.

--- a/lib/rules/header.js
+++ b/lib/rules/header.js
@@ -168,8 +168,9 @@ function genCommentsRange(context, comments, eol) {
  */
 function genPrependFixer(commentType, node, headerLines, eol, numNewlines) {
     return function(fixer) {
-        return fixer.insertTextBefore(
-            node,
+        const insertPos = (node.comments.length && node.comments[0].type === "Shebang") ? node.comments[0].range[1] + eol.length : 0;
+        return fixer.insertTextBeforeRange(
+            [insertPos, 0 /* don't care */],
             genCommentBody(commentType, headerLines, eol, numNewlines)
         );
     };

--- a/tests/lib/rules/header.js
+++ b/tests/lib/rules/header.js
@@ -348,7 +348,6 @@ describe("unix", () => {
                 output: "//Copyright 2020\n//My Company\n\nconsole.log(1);\n//Comment\nconsole.log(2);\n//Comment"
             },
             {
-                name: "Missing empty line after header / windows",
                 code: "//Copyright 2020\r\n//My Company\r\nconsole.log(1);\r\n//Comment\r\nconsole.log(2);\r\n//Comment",
                 options: ["line", ["Copyright 2020", "My Company"], 2, { lineEndings: "windows"}],
                 errors: [
@@ -363,7 +362,55 @@ describe("unix", () => {
                     {message: "no newline after header"}
                 ],
                 output: "//Copyright 2020\n//My Company\n\nconsole.log(1);\n//Comment\nconsole.log(2);\n//Comment"
-            }
+            },
+            {
+                code: "#!/usr/bin/env node\nconsole.log(1);",
+                options: ["line", " Copyright"],
+                errors: [
+                    {message: "missing header"},
+                ],
+                output: "#!/usr/bin/env node\n// Copyright\nconsole.log(1);",
+            },
+            {
+                code: "#!/usr/bin/env node\n\n\nconsole.log(1);",
+                options: ["line", " Copyright"],
+                errors: [
+                    {message: "missing header"},
+                ],
+                output: "#!/usr/bin/env node\n// Copyright\n\n\nconsole.log(1);",
+            },
+            {
+                code: "#!/usr/bin/env node\n\n// My Company\nconsole.log(1);",
+                options: ["line", " Copyright"],
+                errors: [
+                    {message: "missing header"},
+                ],
+                output: "#!/usr/bin/env node\n// Copyright\n\n// My Company\nconsole.log(1);",
+            },
+            {
+                code: "#!/usr/bin/env node\n\n/* Copyright */\nconsole.log(1);",
+                options: ["block", " Copyright "],
+                errors: [
+                    {message: "missing header"},
+                ],
+                output: "#!/usr/bin/env node\n/* Copyright */\n\n/* Copyright */\nconsole.log(1);",
+            },
+            {
+                code: "#!/usr/bin/env node\n/* My Company */\nconsole.log(1);",
+                options: ["block", " Copyright "],
+                errors: [
+                    {message: "incorrect header"},
+                ],
+                output: "#!/usr/bin/env node\n/* Copyright */\nconsole.log(1);",
+            },
+            {
+                code: "#!/usr/bin/env node\n/* Copyright */\nconsole.log(1);",
+                options: ["line", " Copyright"],
+                errors: [
+                    {message: "header should be a line comment"},
+                ],
+                output: "#!/usr/bin/env node\n// Copyright\nconsole.log(1);",
+            },
         ])
     });
 });
@@ -695,7 +742,23 @@ describe("windows", () => {
                     {message: "no newline after header"}
                 ],
                 output: "//Copyright 2020\r\n//My Company\r\n\r\nconsole.log(1);\r\n//Comment\r\nconsole.log(2);\r\n//Comment"
-            }
+            },
+            {
+                code: "#!/usr/bin/env node\r\nconsole.log(1);",
+                options: ["line", " Copyright"],
+                errors: [
+                    {message: "missing header"},
+                ],
+                output: "#!/usr/bin/env node\r\n// Copyright\r\nconsole.log(1);",
+            },
+            {
+                code: "#!/usr/bin/env node\r\n\r\n\r\nconsole.log(1);",
+                options: ["line", " Copyright"],
+                errors: [
+                    {message: "missing header"},
+                ],
+                output: "#!/usr/bin/env node\r\n// Copyright\r\n\r\n\r\nconsole.log(1);",
+            },
         ])
     });
 });

